pkg/oidc/client/client.go

```diff
@@ -205,6 +206,25 @@ func (c *client) verifyToken(ctx context.Context, token *oauth2.Token, nonce str
 	if nonce != "" && nonce != verifiedIDToken.Nonce {
 		return nil, fmt.Errorf("nonce did not match (wants %s but got %s)", nonce, verifiedIDToken.Nonce)
 	}
+
+	if c.useAccessToken {
+		accessToken, ok := token.Extra("access_token").(string)
+		if !ok {
+			return nil, fmt.Errorf("access_token is missing in the token response: %#v", accessToken)
+		}
+		_, err := verifier.Verify(ctx, accessToken)
+		if err != nil {
+			return nil, fmt.Errorf("could not verify the access token: %w", err)
+		}
+
+		// There is no `nonce` to check on the `access_token`. We rely on the
+		// above `nonce` check on the `id_token`.
+
+		return &oidc.TokenSet{
+			IDToken:      accessToken,
+			RefreshToken: token.RefreshToken,
+		}, nil
+	}
```

Comment:
Don't you think it should be enough to select the other token here? 

https://github.com/int128/kubelogin/blob/94ca6bc8d22db04e4c33eac04dd2d53d7a5c89b0/pkg/oidc/client/client.go#L196

The way how it's implemented now would actually lead to a verification of both tokens if `--oidc-use-access-token` is set, which is in my opinion not necessary.

Reply:
That is a valid alternative approach. I chose not do that initially here because the `id_token` has a `nonce` while the `access_token` does not. So if we only validate the `access_token`, we will lose the `nonce` check. It seemed preferred to me to maintain this `nonce` check but use the `access_token`.